### PR TITLE
Prototype: Add App dependencies to dependencies tab

### PR DESCRIPTION
**Pull request by joshraker**
Prototype for finding dependencies in App configurations without DNS resolution. The tests were updated to work but testing more examples e.g. matching vhost `externalHost` and `virtualDomain` would be good in future iterations.

Multiple config keys matching the same resource currently display as multiple rows.

Screenshot 2024-05-03 at 15 59 50

### CODE REVIEW

File: src/deploy/configuration/index.ts [hunk 1]
```diff
@@ -96,63 +102,143 @@ export const selectAppConfigByAppId = createSelector(
...
+
   return deps;
 }
 
-export interface DepGraphDb extends DeployDatabase {
-  why: DepNode;
-}
+// export interface DepGraphDb extends DeployDatabase {
+//   why: DepNode;
+// }
```
Comment (joe-herman) [inline src/deploy/configuration/index.ts:204]: not security related: still necessary?

File: src/deploy/configuration/index.ts [hunk 2]
```diff
@@ -96,63 +102,143 @@ export const selectAppConfigByAppId = createSelector(
...
+    return depGroups;
   },
 );
 
-export interface DepGraphApp extends DeployApp {
-  why: DepNode;
-}
+// export interface DepGraphApp extends DeployApp {
+//   why: DepNode;
+// }
```
Comment (joe-herman) [inline src/deploy/configuration/index.ts:241]: not security related: still necessary?

File: src/secrets/index.ts [hunk 3]
```diff
@@ -0,0 +1,181 @@
+// Generated from https://github.com/gitleaks/gitleaks/blob/79cac73f7267f4a48f4bc73db11e105a6098a836/config/gitleaks.toml
```
Comment (joe-herman) [inline src/secrets/index.ts:1]: question: if we wanted to regenerate this to get the latest set of regexes, how would we do that? Take `regex` items from the toml?

**Review by joe-herman — APPROVED** [review]
Thanks for adding the secrets check!

Nonblocking comments attached

File: src/secrets/index.ts (hunk 3 above)
Comment (joshraker) [reply src/secrets/index.ts:1]: "Derived" may be a more accurate description. I converted from TOML to JSON then it was whack-a-mole to address the JS unsupported regex features like `(?i)` (inline case insensitive flag).

Comment (joshraker) [reply src/secrets/index.ts:1]: If we did want to update it, it'd probably be best to look at the toml diff from the linked commit and patch our regexes accordingly unless there's a lot of changes.

**Comment by joshraker** [comment]
https://app.shortcut.com/aptible/story/24064/prototype-infer-connections-between-aptible-resources

**Review by neurosnap — APPROVED** [review]
nice job!

**Pull request merged**